### PR TITLE
DOC: Fixes the AFQ tract profile tutorial.

### PR DESCRIPTION
**Pull request by arokem**
This also adds the capability to download AFQ derivatives for all of the HBN subjects from the FCP INDI bucket where those are stored as part of the HBN POD2 study.

Addresses #3175, hopefully once and for all.

### CODE REVIEW

**Review by skoudoro — COMMENTED** [review]
Hi @arokem,

Thank you for this.

No sure the reason, but the download was quite long and TQDM did not show up. So I did not know if the terminal was freezing or something was happening. 

Can you look at this, and make sure that TQDM shows up.  Even when the data are already in place, we do not have any information like the other fetcher. 

Otherwise, Can you tell me if this result looks correct to you?
image

Apart from this 2 points, it looks good to go.

**Comment by arokem** [comment]
The result looks correct. I'll look into the tqdm issue and report back.

File: dipy/data/fetcher.py [hunk 1]
```diff
@@ -1980,7 +1980,83 @@ def fetch_hcp(subjects,
...
+
+    # Create the BIDS derivatives description file text
+    to_bids_description(base_dir,
+                        **{"Name": "HBN",
+                           "PipelineDescription": {'Name': 'qsiprep'}})
+
+    return data_files
+
+
+def fetch_hbn(subjects, path=None, include_afq=False):
```
Comment (skoudoro) [inline dipy/data/fetcher.py:2059]: Also, We start to introduce keyword-only arguments in the codebase.

can you replace this by `def fetch_hbn(subjects, *, path=None, include_afq=False):`

Comment (arokem) [reply dipy/data/fetcher.py]: Changed in [5123bcd](https://github.com/dipy/dipy/pull/3178/commits/5123bcd380b07225059cf8c7286c09678b831865). Just noting that I've noticed that this transition is breaking downstream code. I guess folks will have to update all their calls if they want to update dipy.

Comment (skoudoro) [reply dipy/data/fetcher.py]: sorry for that

**Comment by skoudoro** [comment]
Hi @arokem,

> No sure the reason, but the download was quite long and TQDM did not show up. So I did not know if the terminal was freezing or something was happening.

Do you have an update on this

**Comment by skoudoro** [comment]
Hi @arokem, 

It would be great if you could finalize this PR.  It seems very close to go!

Thanks

**Comment by arokem** [comment]
TBH, I am not sure what remains here. You mentioned that it takes a long time to download the data, and TQDM not showing up. The time is because this downloads a large amount of diffusion MRI and derivatives, and I can't replicate TQDM not showing up on my end.

**Comment by arokem** [comment]
Would it be helpful if we download only the bits of the data that are directly used here maybe?  I can see pros and cons to that.

**Comment by skoudoro** [comment]
> Would it be helpful if we download only the bits of the data that are directly used here maybe? I can see pros and cons to that.

I am ok with the long time to download. it was the TQDM issue. 

Let me try again to see if TQDM shows up.

**Comment by skoudoro** [comment]
Can you also address the code format issue. See https://github.com/dipy/dipy/actions/runs/8715096640/job/23906448911?pr=3178#step:4:130

This is just spacing issue. Thanks !

File: doc/examples/afq_tract_profiles.py [hunk 2]
```diff
@@ -31,44 +29,58 @@
...
-bundles_path = dpd.fetch_bundles_2_subjects()
-bundles_folder = bundles_path[1]
+subject = "NDARAA948VFH"
+session = "HBNsiteRU"
 
-cst_l_file = op.join(bundles_folder, "bundles_2_subjects", "subj_2", "bundles",
-                     "bundles_cst.left.trk")
-af_l_file = op.join(bundles_folder, "bundles_2_subjects", "subj_2", "bundles",
-                    "bundles_af.left.trk")
+fdict, path = fetch_hbn([subject], include_afq=True)
```
[the remark this answers is not in the document]
Comment (arokem) [reply doc/examples/afq_tract_profiles.py:42]: Fixed in last series of commits

**Comment by arokem** [comment]
Now rebased on top of main and with ruff formatting applied.

**Comment by skoudoro** [comment]
Thank you for the update @arokem. Merging since codespell and code format are ok, we do not have CI's for examples yet. 
something to fix in a close future.
